tests/dind.sh

```diff
@@ -1,15 +1,20 @@
-#!/bin/sh
+#!/bin/bash
 
+K3D_EXE="/bin/k3d"
 K3D_IMAGE_TAG=$1
 
 k3de2e=$(docker run -d --rm \
           -v "$(pwd)"/tests:/tests \
           --privileged \
-          -e EXE="/bin/k3d" \
+          -e EXE="$K3D_EXE" \
           -e CI="true" \
+          --name "k3d-e2e-runner" \
           k3d:$K3D_IMAGE_TAG)
 
 sleep 5 # wait 5 seconds for docker to start
 
 # Execute tests
-docker exec $k3de2e /tests/runner.sh
\ No newline at end of file
+finish() { docker stop "$k3de2e" || /bin/true ; }
```

Comment:
We could as well remove it upon successful completion, because one probably only would want to retain the stopped container to inspect logs (i.e. if it failed), right? :thinking:

Reply:
I will add a variable for (optionally) keeping the e2e runner.